src/Tests/AzureClientEntryPointTests.cs

```diff
@@ -81,6 +82,94 @@ public async Task FromSnippet()
             Assert.IsNotNull(job);
         }
 
+        [TestMethod]
+        public async Task QIRSubmission()
+        {
+            var entryPointGenerator = Init("Workspace", new string[] { SNIPPETS.HelloQ });
+            var entryPoint = entryPointGenerator.Generate("HelloQ", null, forceQir: true);
+
+            Assert.IsNotNull(entryPoint);
+            var job = await entryPoint.SubmitAsync(
+                new MockQIRSubmitter(new List<Argument>()),
+                new AzureSubmissionContext());
+            Assert.IsNotNull(job);
+        }
+
+        [TestMethod]
+        public async Task ValidParameterTypes()
+        {
+            var entryPointGenerator = Init("Workspace", new string[] { SNIPPETS.ValidParameterTypes });
+            var entryPoint = entryPointGenerator.Generate("UseValidParameterTypes", null, forceQir: true);
+
+            Assert.IsNotNull(entryPoint);
+
+            var validArguments = new Dictionary<Argument, string>()
+            {
+                { new Argument("myBool", new ArgumentValue.Bool(true)), "\"true\"" },
+                { new Argument("myDouble", new ArgumentValue.Double(1.2)), "\"1.2\"" },
+                { new Argument("myInt", new ArgumentValue.Int(2)), "\"2\"" },
+                { new Argument("myStr", new ArgumentValue.String("\"Hello\"")), "\"Hello\"" },
+                { new Argument("myPauli", new ArgumentValue.Pauli(Pauli.PauliX)), "\"PauliX\"" },
+                { new Argument("myResult", new ArgumentValue.Result(Result.One)) , "\"1\"" },
+                { new Argument("innerInt", new ArgumentValue.Int(7)) , "\"7\"" },
+                { new Argument("innerDouble", new ArgumentValue.Double(6.4)) , "\"6.4\"" }
+            };
+
+            var job = await entryPoint.SubmitAsync(
+                new MockQIRSubmitter(validArguments.Select(x => x.Key).ToList()),
+                new AzureSubmissionContext()
+                {
+                    InputParameters = validArguments.ToDictionary(x => x.Key.Name, x => x.Value)
+                });
+            Assert.IsNotNull(job);
+        }
+
+        [TestMethod]
+        public async Task InvalidParameterTypes()
```

Comment:
Can you please add test cases for operations that have a mix of valid and invalid parameter types?

Reply:
The "UseEmbeddedUnitType" Snippet case should test that, I believe. It is part of the InvalidParameterTypes test. Was there something different you had in mind?

Reply:
It would be great to add similar cases but with the other two unsupported types (array and range). For these new cases it would also be great to position the unsupported type at the beginning and end of the list just to add a bit of coverage on the position of arguments too.

Reply:
done